elements/pf-date-picker/pf-date-picker.ts

```diff
@@ -0,0 +1,599 @@
+      this.inputDate = new Date(this.inputDateWithUniqueTimeStamp.split('#')[0]);
+      this.#setInputDate();
+    }
+
+    if (changedProperties.has('placeholderTextWithUniqueCode')) {
+      const [placeholder, uniqueCode] = this.placeholderTextWithUniqueCode.split('#');
+      this.dateFormat = placeholder;
+    }
+  }
+
+  render() {
+    const invalidIconClasses = { isDateInvalid: !this.isDateValid, hidden: this.isDateValid };
+    const invalidTextClasses = { showInvalidText: !this.isDateValid, hidden: this.isDateValid };
+    const invalidInputClasses = { invalidDateInput: !this.isDateValid };
+
+    return html`
+      <div class="date-picker-container">
+        <div class="datepicker">
+          <div class="date-input-box">
+            <div class="date-input-box-container">
+              <input 
+                type="text" 
+                aria-label="Date picker"
+                @input=${this.#onInput} 
+                .value=${this.formattedDate ?? ''} 
+                @click=${this.#dateInputClick}
+                id="date-input" 
+                .disabled=${this.isDisabled}
+                placeholder=${this.dateFormat} 
+                class=${classMap(invalidInputClasses)}
+                part="input"
+              />
+              <pf-icon class=${classMap(invalidIconClasses)} size="md" icon="exclamation-circle" set="fas"></pf-icon>
+              <div class=${classMap(invalidTextClasses)}> ${this.errorMessage} </div>
+            </div>
+            <pf-popover class=${this.isDisabled && 'disable-popover'} id="popover" part="popover" hide-close no-outside-click>
+              <div part="toggle" class="date-picker-toggle-container">
+                <pf-button 
+                  label="Toggle date picker" 
+                  .disabled=${this.isDisabled} 
+                  id="datePickerToggleButton" 
+                  @click=${this.#openPopover} 
+                  variant="control" 
+                  part="toggle-button"
+                  class="date-picker-calendar-icon">
+                    <pf-icon part="icon" icon="calendar-alt" set="far" size="md"></pf-icon>
+                </pf-button>
+              </div>
+              <div slot="body">
+                <div class="calendar" id="calendarWrapper">
+                  <div class="date-picker-table" id="datePickerTableContent">
+                    <div class="date-picker-table-row date-picker-table-main-header">
+                      <div class="date-picker-table-col">
+                        <pf-previous-button 
+                          .currentYear=${this.currentYear} 
+                          .currentMonth=${this.currentMonthIndex} 
+                          @previousMonthAndYear=${this.#getPreviousMonthAndYear}>
+                        </pf-previous-button>
+                      </div>
+                      <div class="date-picker-table-col date-picker-table-month-year">
+                        <pf-month-select 
```

Comment:
Hey @zeroedin 
Thank you so much for reviewing.

I developed the <pf-month-select> component for the date-picker because <pf-dropdown> was not yet available. I will go ahead and change it now to <pf-dropdown> as it will be a much simpler implementation.